### PR TITLE
feat: Add search bar on user profile page.

### PR DESCRIPTION
**Pull request by appleboy**
Screenshot as below:

screen shot 2017-01-29 at 8 57 31 pm

* Sort and search keyword on repository list

### CODE REVIEW

**Comment by lunny** [comment]
LGTM

**Comment by lunny** [comment]
conflicted

**Comment by appleboy** [comment]
@lunny Done.

File: routers/user/profile.go [hunk 1]
```diff
@@ -109,12 +109,66 @@ func Profile(ctx *context.Context) {
...
+				Page:     page,
+				PageSize: setting.UI.User.RepoPagingNum,
+			})
+			if err != nil {
+				ctx.Handle(500, "SearchRepositoryByName", err)
+				return
+			}
+
+			ctx.Data["Repos"] = repos
+			ctx.Data["Page"] = paginater.New(int(count), setting.UI.User.RepoPagingNum, page, 5)
```
Comment (strk) [inline routers/user/profile.go:160]: What is the hard-coded `5` here ?

Comment (strk) [reply routers/user/profile.go:160]:
```
$ go doc github.com/Unknwon/paginater.New
func New(total, pagingNum, current, numPages int) *Paginater
    New initialize a new pagination calculation and returns a Paginater as
    result.
```

What's the meaning of `numPages` ? How's that different from `total`/`pagingNum` ?
@Unknwon ? Maybe the documentation on that could be improved

Comment (strk) [reply routers/user/profile.go:160]: From the README.md of the paginater:
```
        // Arguments:
        // - Total number of rows
        // - Number of rows in one page
        // - Current page number 
        // - Number of page links
        p := paginater.New(45, 10, 3, 3)
```

I still don't understand what `Number of page links` means though.
Is it the expected number of pages ? As that could be computed as ceil(`Total number of rows` / `Number of rows in one page`), right ?

Comment (strk) [reply routers/user/profile.go:160]: Ok I think I got it: it's the number of page links you want to be visible in a template. Sounds a bit wrong to mix output control and flow control to me, but this is not the place to discuss this. Still maybe it's good to have a configuration in settings about the number of pager links ? As I see that hard-coded `5` in many places even only in this file...

**Review by strk — APPROVED** [review]
Sounds good to me, beside the desire of see the hard-coded "number of page links" pager parameter be replaced by a system configuration.

**Comment by strk** [comment]
LGTM, even :)

File: routers/user/profile.go (hunk 1 above)
Comment (appleboy) [reply routers/user/profile.go:160]: @strk I think this may be a bug. I change the `Number of page links` to `2`. It still show five items in a template.

**Comment by lunny** [comment]
let L-G-T-M work

**Pull request merged**